astropy/coordinates/representation.py

```diff
@@ -1596,21 +1898,39 @@ def to_cartesian(self, base):
         base : instance of ``self.base_representation``
              The points for which the differentials are to be converted: each of
              the components is multiplied by its unit vectors and scale factors.
+
+        Returns
+        -------
+        This object as a `CartesianDifferential`
         """
-        base_e, base_sf = self._get_base_vectors(base)
-        return functools.reduce(
-            operator.add, (getattr(self, d_c) * base_sf[c] * base_e[c]
-                           for d_c, c in zip(self.components, base.components)))
+        return self._to_cartesian_helper(base)
 
     @classmethod
     def from_cartesian(cls, other, base):
-        """Interpret 3D rectangular cartesian coordinates as differentials.
+        """Convert the differential from 3D rectangular cartesian coordinates to the desired class.
 
         Parameters
         ----------
+        other :
+            The object to convert into this differential.
         base : instance of ``self.base_representation``
-            Base relative to which the differentials are defined.
+             The points for which the differentials are to be converted: each of
+             the components is multiplied by its unit vectors and scale factors.
+
+        Returns
+        -------
+        A new differential object that is this class' type.
         """
+        return cls._from_cartesian_helper(other, base)
+
+    def _to_cartesian_helper(self, base):
```

Comment:
I think this function and ```_from_cartesian_helper``` need fuller docstrings to explain to those less familiar with the concepts behind representation of differentials, perhaps with a spherical coordinate or cylindrical coordinate example?